matlab/schemas/+map/OptImageBar.m

```diff
@@ -20,69 +20,61 @@
                     
+                    % get frame times
+                    if ~exist(stimulus.Sync & key)
+                        disp 'synchronizing...'
+                        populate(stimulus.Sync,key)
+                    end
+                    frame_times =fetch1(stimulus.Sync & key,'frame_times');
+
                     % get the vessel image
                     disp 'getting the vessels...'
                     k = [];
                     k.session = key.session;
                     k.animal_id = key.animal_id;
                     k.site_number = fetch1(experiment.Scan & key,'site_number');
-                    vesObj = experiment.Scan & k & 'software = "imager" and aim = "vessels"';
+                    vesObj = experiment.Scan - experiment.ScanIgnored & k & 'software = "imager" and aim = "vessels"';
                     if ~isempty(vesObj)
-                        name = fetch1( vesObj ,'filename');
-                        if isempty(strfind(name,'.h5')); name = [name '.h5'];end
+                        names = fetchn( vesObj ,'filename');
+                        name = names{end};
+                        if ~contains(name,'.h5'); name = [name '.h5'];end
                         filename = getLocalPath(fullfile(path,name));
                         vessels = squeeze(mean(getOpticalData(filename)));
                     end
                     
                 case 'scanimage'
                     
-                    % get Optical data
+                   % get Optical data
                     disp 'loading movie...'
-                    reader = preprocess.getGalvoReader(key);
-                    Data = squeeze(mean(reader(:,:,1,:,:),4));
-                    Data = permute(Data,[3 1 2]);
-                    [nslices, data_fs] = fetch1(preprocess.PrepareGalvo & key,'nslices','fps');
+                    if strcmp(setup,'2P4') % mesoscope
+                        path = getLocalPath(fullfile(path, sprintf('%s*.tif', name)));
+                        reader = ne7.scanreader.readscan(path,'int16',1);
+                        Data = permute(squeeze(mean(reader(:,:,:,:,:))),[3 1 2]);
+                        data_fs = reader.fps;
+                        nslices = length(reader.fieldDepths);
+                    else
+                        reader = preprocess.getGalvoReader(key);
+                        Data = squeeze(mean(reader(:,:,1,:,:),4));
+                        Data = permute(Data,[3 1 2]);
+                        [nslices, data_fs] = fetch1(preprocess.PrepareGalvo & key,'nslices','fps');
+                    end
                     
                     % calculate frame times
-                    frame_times = fetch1(preprocess.Sync & key,'frame_times');
+                    frame_times = fetch1(stimulus.Sync & key,'frame_times');
                     frame_times = frame_times(1:nslices:end);
+                    frame_times = frame_times(1:size(Data,1));
```

Comment:
Not sure why the double downsampling of frameTimes here, I thought downsampling using nSlices (proper slices no depths) should be enough

Reply:
If the scan did not complete, not all fields have the same number and the frame_times is longer